internal/sftp/service.go

```diff
@@ -0,0 +1,9 @@
+package sftp
+
+import (
+	"io"
+)
+
+type Service interface {
+	Upload(src io.Reader, dest string) (int64, error)
+}
```

Comment:
I've used `Service` a lot in other packages when I couldnt' find a better name. In this particular case maybe one name that could work better is `Client`.

Reply:
Suggested docstring:

```go
// A Client manages the transmission of data over SFTP.
//
// Implementations of the Client interface handle the connection details,
// authentication, and other intricacies associated with different SFTP
// servers and protocols.
type Client interface {
	// Upload transfers data from the provided source reader to a specified
	// destination on the SFTP server.
	Upload(src io.Reader, dest string) (int64, error)
}
```

Reply:
Man naming, right? :weary: 

TL;DR `Service` -> `Client` sounds fine. :)

I'm still debating whether it would be better to call this package `upload` and leave the upload protocol up to the implementation (e.g. SFTP, SCP, Blob?).  But then what do you call the upload method, `upload.Do()`?  Plus all of the code is implementing SSH/SFTP, so it's probably better to call a spade a spade.

It does some weird to me to have an sftp service with only an `Upload()` method. One advantage of the current design is that it could support other sftp methods like download, or public dial and close methods to use an SFTP session to upload multiple files. :shrug:

Reply:
You're right. This is all reminded me to Douglas' work where he used sftpblob, a driver for [gocloud.dev/blob] for SFTP where the concept of blob is pushed beyond traditional S3 buckets, see https://github.com/artefactual-sdps/enduro/blob/735afd570567ab609519496d45822819d3daebd4/internal/storage/types/location_config.go.

Reply:
I considered using https://github.com/rukavina/sftpblob like in Douglas' code, but it looks like a personal project that was abandoned years ago.  I didn't look around for other SFTP drivers for gocloud.dev/blob but I sure didn't want to write one myself. :laughing: 

I don't know if your suggesting changing this code to something more like Douglas' blob implementation, but if you are I'm not sure how to get there from here. :thinking:

Reply:
It could be worth the effort but definitely not in this iteration, it's a lot of work to write a blob Driver I think.